### DISCUSSION TITLE
How to upload multiple static files to the Wave App while preserving their hierarchy?

### DISCUSSION THREAD

**Discussion by nipunchamikara** [h2oai/wave#2321] [Q&A] [answered]
Rather than using a CDN to fetch JS files of a library (Monaco Editor), I want it inside the Wave app as static files. `ui.upload` requires the file path of each file, and there are too many of these files. Since these files may depend on each other, I want to make sure the hierarchy of these files is maintained. Is it possible to do this in Wave?

**Comment by mturoci** [comment, marked as answer]
There is `q.site.upload_dir` for this exact scenario. See https://wave.h2o.ai/docs/files/#provide-file-downloads

<img width="1084" alt="image" src="https://github.com/h2oai/wave/assets/64769322/4bd5de60-2550-432b-8eda-a486280c371a">
